environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -621,10 +629,20 @@ public Weld addAlternativeStereotype(Class<? extends Annotation> alternativeSter
      * @see #ARCHIVE_ISOLATION_SYSTEM_PROPERTY
      * @see #SHUTDOWN_HOOK_SYSTEM_PROPERTY
      * @see #DEV_MODE_SYSTEM_PROPERTY
+     * @see #ADDITIONAL_BEAN_DEFINING_ANNOTATIONS_PROPERTY
      * @see ConfigurationKey
      */
     public Weld property(String key, Object value) {
         properties.put(key, value);
+        if (ADDITIONAL_BEAN_DEFINING_ANNOTATIONS_PROPERTY.equals(key) && value instanceof Collection) {
+            for (Object element : ((Collection<?>)value)) {
+                if (element instanceof Class && Annotation.class.isAssignableFrom((Class<?>) element)) {
```

Comment:
I think it would make sense to blow up in an `else` block if these conditions aren't met - fail fast.

Reply:
I went back and forth on this.  If I supply "`cheese`" for one of the System properties in this class that is expecting `true` or `false`, nothing blows up.  I followed suit here.  WDYT?

Reply:
Hmm, good point.
OTOH checking true/false is much simpler and you always have a default value. In this case you can have three scenarios - all correct, all wrong, a corrupted list where some are good and some are bad. And you have no fallback values, do you think it makes sense to silently proceed?

Or we could add some INFO logging for all wrong values that user tried to register perhaps?